File: eis-o-meter/Api/src/EisoMeter.Api/Features/Claims/GetClaimStatus/Contracts.cs
namespace EisoMeter.Api.Features.Claims.GetClaimStatus;


public sealed record GetClaimStatusParameter(
    [property: JsonPropertyName("email")] string Email = ""
);

public sealed record GetClaimStatusResponse(
    [property: JsonPropertyName("lastUpdate")] string LastUpdate,
    [property: JsonPropertyName("highestTemperature")] double HighestTemperature,
    [property: JsonPropertyName("status")] GetClaimStatusType Status
);

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum GetClaimStatusType
{
    Locked,
    AlreadyClaimed,
    ReadyToClaim
}

File: eis-o-meter/Ui/src/feature/ice-reward/services/Contracts.ts
export interface GetClaimStatusResponse {
    lastUpdate: string;
    highestTemperature: number;
    status: GetClaimStatusType;
}

export enum GetClaimStatusType {
    Locked = "Locked",
    AlreadyClaimed = "AlreadyClaimed",
    ReadyToClaim = "ReadyToClaim",
}

export interface ClaimRewardResponse {
    status: ClaimRewardType;
}

export enum ClaimRewardType {
    SuccessfulClaimed = "SuccessfulClaimed",
    Expired = "Expired",
}